### PR TITLE
Feat/conect room page with back end

### PR DESCRIPTION
**Pull request by DeidamiaFuentes**
In this PR I connected the Select Room page to the backend to get the different rooms from the hotel

### CODE REVIEW

File: frontend/src/pages/SelectRoom.jsx [hunk 1]
```diff
@@ -1,62 +1,57 @@
...
-    FloorNumber: 2,
-    Code: "ijk",
-    Services: ["b", "c"]
-  }]
+    }
+    fetchRooms();
+  }, [checkInDate, checkOutDate]);
+
+  if (loading) {
+    return <div>Loading...</div>;
```
Comment (MayerliSantander) [inline frontend/src/pages/SelectRoom.jsx]: This loading should appear in the center and slightly larger, with a noticeable font for a better user experience.

File: frontend/src/pages/SelectRoom.jsx [hunk 2]
```diff
@@ -1,62 +1,57 @@
...
 
-  const rooms = [{
-    Beds: [
-      {
-        Size: "Queen",
-        Capacity: "2"
+  useEffect(() => {
+    async function fetchRooms() {
+      try {
+        const availableRooms = await getAvailableRooms(checkInDate, checkOutDate);
```
Comment (MayerliSantander) [inline frontend/src/pages/SelectRoom.jsx:17]: If there are no rooms available on the requested dates, a message should be displayed warning of this and to try other dates, it should be in the middle as well as the loading.

**Review by MayerliSantander — CHANGES_REQUESTED** [review]

**Review by MayerliSantander — CHANGES_REQUESTED** [review]
Captura de pantalla de 2024-10-15 22-30-09
The room text is not centered on the defined resolution.

File: frontend/src/pages/SelectRoom.jsx (hunk 1 above)
Comment (DeidamiaFuentes) [reply frontend/src/pages/SelectRoom.jsx]: Done :D

File: frontend/src/pages/SelectRoom.jsx (hunk 2 above)
Comment (DeidamiaFuentes) [reply frontend/src/pages/SelectRoom.jsx:17]: Done :D

**Review by MayerliSantander — APPROVED** [review]

**Review by OropezaHugo — APPROVED** [review]

**Pull request merged**